internal/util/K8sUtil.go

```diff
@@ -251,3 +266,88 @@ func (impl K8sUtil) UpdateSecret(namespace string, secret *v1.Secret, client *v1
 		return secret, nil
 	}
 }
+
+func (impl K8sUtil) DeleteJobIfExists(namespace string, name string, clusterConfig *ClusterConfig) error {
+	clientSet, err := impl.GetClientSet(clusterConfig)
+	if err != nil {
+		impl.logger.Errorw("clientSet err, DeleteJobIfExists","err", err)
+		return err
+	}
+	jobs := clientSet.BatchV1().Jobs(namespace)
+
+	job, err := jobs.Get(name, metav1.GetOptions{})
+	if err != nil {
+		impl.logger.Errorw("get job err, DeleteJobIfExists","err", err)
+		return nil
+	}
+
+	if job != nil {
+		err := jobs.Delete(name, &metav1.DeleteOptions{})
+		if err!=nil && errors.IsNotFound(err){
+			impl.logger.Errorw("delete err, DeleteJobIfExists","err", err)
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (impl K8sUtil) CreateJob(clusterConfig *ClusterConfig, namespace string, job *batchV1.Job, name string) error {
+	clientSet, err := impl.GetClientSet(clusterConfig)
+	if err != nil {
+		impl.logger.Errorw("clientSet err, CreateJob","err", err)
+	}
+	time.Sleep(5 * time.Second)
+
+	jobs := clientSet.BatchV1().Jobs(namespace)
+	_, err = jobs.Get(name, metav1.GetOptions{})
+	if err == nil {
+		impl.logger.Errorw("get job err, DeleteJobIfExists","err", err)
+		time.Sleep(5 * time.Second)
+		_, err = jobs.Get(name, metav1.GetOptions{})
+		if err == nil{
+			return error2.New("Job deletion takes more time than expected, please try after sometime")
+		}
+	}
+
+	_ , err = jobs.Create(job)
+	if err != nil {
+		impl.logger.Errorw("create err, CreateJob","err", err)
+		return err
+	}
+	return nil
+}
+
+func (impl K8sUtil) CreateJobSafely(content []byte, namespace string, clusterConfig *ClusterConfig) error {
```

Comment:
CreateJobSafely -> DeleteAndCreateJob 

//Deletes and recreates if job exists else creates the job